tekton/ci/interceptors/github/pkg/github/issue_comment.go

```diff
@@ -0,0 +1,109 @@
+package github
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/google/go-github/v34/github"
+	"github.com/tektoncd/plumbing/tekton/ci/interceptors/github/pkg/github/bindings"
+	pb "github.com/tektoncd/plumbing/tekton/ci/interceptors/github/pkg/proto/v1alpha1/config_go_proto"
+	"github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+	"google.golang.org/grpc/codes"
+)
+
+var (
+	_ = Interceptor(&IssueComment{})
+)
+
+type IssueComment struct{}
+
+func (c *IssueComment) Execute(ctx context.Context, client *github.Client, cfg *pb.Config, req *v1alpha1.InterceptorRequest) (*v1alpha1.InterceptorResponse, error) {
+	event := new(github.IssueCommentEvent)
+	if err := json.Unmarshal([]byte(req.Body), event); err != nil {
+		return nil, Errorf(codes.InvalidArgument, err.Error())
+	}
+
+	if event.GetAction() != "created" {
+		return nil, Errorf(codes.Unimplemented, "unsupported action")
+	}
+
+	commentCfg := cfg.GetPullRequest().GetComment()
+	if commentCfg == nil {
+		// No approver config - take no action (should be covered by PR handler)
+		return nil, Error(codes.FailedPrecondition, "comment config not enabled")
+	}
+
+	// Check if comment matches.
+	match := commentCfg.GetMatch()
+	if match == "" {
+		match = "/ok-to-test"
+	}
+	re, err := regexp.Compile(match)
+	if err != nil {
+		return nil, Error(codes.FailedPrecondition, "invalid match keyphrase")
+	}
+	if !re.MatchString(event.GetComment().GetBody()) {
+		return nil, Error(codes.FailedPrecondition, "comment does not match keyphrase")
+	}
+
+	// See if the comment came from an approved user. We do this after the
+	// comment match to save an API call if we can.
+	approverCfg := commentCfg.GetApprovers()
+	path := approverCfg.GetPath()
+	if path == "" {
+		path = "OWNERS"
+	}
+	org := event.GetRepo().GetOwner().GetLogin()
+	repo := event.GetRepo().GetName()
+	number := event.GetIssue().GetNumber()
+	commentAuthor := event.GetComment().GetUser().GetLogin()
+
+	fc, _, _, err := client.Repositories.GetContents(ctx, org, repo, path, &github.RepositoryContentGetOptions{Ref: approverCfg.GetRevision()})
```

Comment:
Am I understanding right, this gets the OWNERS file from the pull request itself? Shouldn't be from the main branch?  (or as configured in github.DefaultBranch from the pr event)

Reply:
Not quite - this gets the revision to fetch from the interceptor config. If the revision is empty, GitHub uses the repo's default branch.
We avoid using the pull request revision to prevent users from hijacking the config.